DuckDuckGo/SpeechRecognizer.swift

```diff
@@ -21,18 +21,48 @@ import Foundation
 import Speech
 import Accelerate
 
-class SpeechRecognizer: SpeechRecognizerProtocol {
-    private var audioEngine = AVAudioEngine()
+protocol SpeechRecognizerDelegate: AnyObject {
+    func speechRecognizer(_ speechRecognizer: SpeechRecognizer, availabilityDidChange available: Bool)
+}
+
+class SpeechRecognizer: NSObject, SpeechRecognizerProtocol {
+    weak var delegate: SpeechRecognizerDelegate?
+    private var audioEngine: AVAudioEngine?
     private var recognitionRequest: SFSpeechAudioBufferRecognitionRequest?
     private var recognitionTask: SFSpeechRecognitionTask?
-    private let speechRecognizer = SFSpeechRecognizer()
+    private let speechRecognizer: SFSpeechRecognizer?
+    private let operationQueue: OperationQueue
     
-    var isAvailable: Bool {
-        // https://app.asana.com/0/1201011656765697/1201271104639596
-        if #available(iOS 15.0, *) {
-            return (speechRecognizer?.isAvailable ?? false) && supportsOnDeviceRecognition
-        } else {
-            return false
+    private(set) var isAvailable = false {
+        didSet {
+            delegate?.speechRecognizer(self, availabilityDidChange: isAvailable)
+        }
+    }
+    
+    override init() {
+        operationQueue = OperationQueue()
+        operationQueue.qualityOfService = .userInteractive
+        
+        speechRecognizer = SFSpeechRecognizer()
+        speechRecognizer?.queue = operationQueue
```

Comment:
I can't do that because they are let properties so they need to be initialized inside the init, and if I change to lazy they would not be thread safe anymore.

I don't think it's worth to do these changes given that the current initialization of those properties are fairly simple and make contextual sense with init method.